main.go

```diff
@@ -79,12 +83,36 @@ func main() {
 		os.Exit(1)
 	}
 
+	operatorInstallationNamespace, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		setupLog.Error(err, "Failed to retrieve operator namespace")
+		os.Exit(1)
+	}
+
+	// If a watch namespace is detected (i.e. operator is namespace scoped), then pass the NS to cache.Options.DefaultNamespaces
+	// If no watch namespace is detected (i.e. operator is cluster scoped), then pass an empty Cache object
+	var managerCache = cache.Options{}
+	if namespace != "" {
+		managerCache = cache.Options{
+			ByObject: map[client.Object]cache.ByObject{
+				&corev1.ConfigMap{}: {
+					Namespaces: map[string]cache.Config{
+						operatorInstallationNamespace: {},
```

Comment:
are you sure we want this? Can you explain what it means?

Reply:
@MStokluska ,  in sigs.k8s.io/controller-runtime **v0.17.2** , **Options** struct  in `NewManager func(config *Config, options Options) ` already doesn't have **Namespace** field. So we are define Namespace in Cache.
Actually if I remember, it was your idea to place Namespace to cache. And seems it's better, than my previouse implementation where Namespace was added to watch in controller. Seems to me now it looks good.

Reply:
We had a call to discuss this, the configMap in cache isn't necessary, same as for operator installation namespace. 
Instead, we are relying on:
- WATCH_NAMESPACE  = "" -> cluster scoped
- WATCH_NAMESPACE != "" -> cache.namespaces must include watched namespace. 

Thanks for update Valery.

Reply:
Done, as discussed. Thank you Michal